Source/Utility/Diagnostics/DiagProbe.cpp

```diff
@@ -0,0 +1,324 @@
+#include "DiagProbe.H"
+#include <AMReX_VisMF.H>
+#include <AMReX_FPC.H>
+#include <AMReX_PlotFileUtil.H>
+#include <regex>
+#include <cstdio>
+
+AMREX_GPU_DEVICE
+AMREX_FORCE_INLINE
+amrex::Real
+LinearInterpolate(
+  amrex::GpuArray<amrex::Real, AMREX_SPACEDIM> xp,
+  amrex::GpuArray<amrex::Real, AMREX_SPACEDIM> x_low,
+  amrex::Array3D<amrex::Real, 0, 1, 0, 1, 0, 1> cell_data,
+  amrex::GpuArray<amrex::Real, AMREX_SPACEDIM> dx)
+{
+  amrex::Real alpha, beta, gama;
+  amrex::Real value = 0.0;
+
+  alpha = 0.0;
+  beta = 0.0;
+  gama = 0.0;
```

Comment:
all this should be defined as one line. And use ternary expressions for beta and (renamed?) gamma.

Reply:
Removed the variables alpha, beta and gama and replaced with an array from the code snippet you suggested.